contacts/urls.py

```diff
@@ -2,6 +2,10 @@
 
 from . import views
 
+
+app_name = 'contacts'
 urlpatterns = [
-    url(r'^$', views.index, name='index')
+    url(r'^$', views.ContactView.as_view(), name='contact'),
+    # url(r'^(?P<pk>[0-9]+)/$', views.DetailView.as_view(), name='detail'),
+    url(r'^([0-9]+)/$', views.detail, name='detail')
```

Comment:
the ending slash should be optional in url.

Reply:
done